File: owen/hash/hash_P1.py
# 완주하지 못한 선수

import collections

def solution(participant, completion):

    answer = list(collections.Counter(participant) - collections.Counter(completion))[0]
    
    return answer



# 정확성  테스트
# 테스트 1 〉	통과 (0.06ms, 10.1MB)
# 테스트 2 〉	통과 (0.04ms, 10.2MB)
# 테스트 3 〉	통과 (0.24ms, 10.3MB)
# 테스트 4 〉	통과 (0.49ms, 10.3MB)
# 테스트 5 〉	통과 (0.45ms, 10.4MB)
# 효율성  테스트
# 테스트 1 〉	통과 (27.55ms, 24.2MB)
# 테스트 2 〉	통과 (51.17ms, 27.7MB)
# 테스트 3 〉	통과 (54.32ms, 30MB)
# 테스트 4 〉	통과 (71.58ms, 39MB)
# 테스트 5 〉	통과 (73.01ms, 39MB)

File: kujin-CSharp/Hash/hash_p1.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> map;

    for(vector<int>::size_type i =0; i<participant.size(); i++){
        map[participant[i]] += 1;
    }
    for(vector<int>::size_type i =0; i<completion.size(); i++){
        map[completion[i]] -= 1;
    }
    for(vector<int>::size_type i =0; i<participant.size(); i++){
        if(map[participant[i]] != 0)
        {
            answer = participant[i];
            break;
        }
    }


    return answer;
}